src/Embed.php

```diff
@@ -15,7 +15,8 @@ class Embed implements CopilotTag
 
     public function __construct($uri, $subtype = EmbedSubtype::IFRAME, $caption = "")
     {
-        if (!is_string($uri)) throw new \InvalidArgumentException('Embed::__construct first argument $uri must be a string. Given:'.($uri?" $uri":'').' ('.gettype($uri).')');
+        if (!is_string($uri)) throw new \InvalidArgumentException("Embed::__construct first argument \$uri must be a string. Given: ".($uri ? "$uri " : "")."(".gettype($uri).").");
+        if (preg_match('/\s/', $uri)) throw new \InvalidArgumentException("Embed::__construct first argument \$uri cannot contain whitespace. Given: \"".str_replace("\n", "\\n", $uri)."\".");
```

Comment:
What are the pros/cons of this? We could allow urls with leading/trailing whitespace and just trim them in the library?

Reply:
@amillward In tugboat, we removed any newlines and then trimmed leading/trailing spaces. Removing newlines and continuing silently feels wrong here, but allowing leading/trailing spaces seems fine, and it saves them trimming the input. I'll add this now.